### DISCUSSION TITLE
How to hook into Rector's lifecycle?

### DISCUSSION THREAD

**Discussion by ghost** [rectorphp/rector#7406] [Q&A] [answered]
I'm writing my own Rector. Besides changing Nodes I also collect some data that I want to use after rector finishes refactoring. Does rector have any events I can hook into? I've already tried to register an event subscriber on `ConsoleEvents::TERMINATE` in rector.php config file, but it does not fire up. However even if it did I wonder how does it play with parallel processing, i.e. I want to execute my code when all the refactoring is done, not when some parallel worker finishes.

**Comment by TomasVotruba** [comment, marked as answer]
Hi, there are no events. It is based on reactjs and [ndjsong](http://ndjson.org/).

**Reply by ghost** [reply]
Ok. Do you have another idea how can I accomplish my goal?

**Reply by TomasVotruba** [reply]
Not really.

**Reply by ghost** [reply]
Ok then. I will have to find some primitive workaround and also disable parallel run.
